lib/moon/helpers/surface_render.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Moon.Helpers.SurfaceRender do
+  @moduledoc """
+  Functions for rendering Surface components in HEEX/SLIM/... (not Surface) templates
+  """
+
+  import Phoenix.LiveView.Helpers, only: [component: 2, live_component: 1]
+
+  defp get_default_props(module) do
+    Enum.reduce(module.__props__(), %{__context__: %{}}, fn
+      %{name: name, opts: opts}, acc ->
+        Map.put(acc, name, Keyword.get(opts, :default))
+    end)
+  end
+
+  defp transform_slots(
+         props = %{inner_block: [%{__slot__: :inner_block, inner_block: inner_block}]}
+       ) do
+    props
+    |> Map.delete(:inner_block)
+    |> Map.put(:default, [%{__slot__: :default, inner_block: inner_block}])
+  end
+
+  defp transform_slots(props), do: props
+
+  defp render_stateless_component(props = %{module: module}) do
+    component(
+      &module.render/1,
+      module
+      |> get_default_props()
+      |> Map.merge(props |> Map.delete(:module) |> transform_slots())
+    )
+  end
+
+  defp render_stateful_component(props = %{module: module}) do
+    module |> get_default_props() |> Map.merge(props |> transform_slots()) |> live_component()
+  end
+
+  defp get_render_function(module) do
+    %{kind: :component} = module.__live__()
```

Comment:
This should not be "black magic". Throught LiveViews these are separate consepts.

Reply:
both methods - with & without magic are supported and described in usage_page.ex